src/main/java/com/demo/MybatisApplication/service/StudentService.java

```diff
@@ -1,51 +1,76 @@
 package com.demo.MybatisApplication.service;
 
-import com.demo.MybatisApplication.dto.StudentAddDto;
-import com.demo.MybatisApplication.dto.StudentDisplayAsSubjects;
-import com.demo.MybatisApplication.dto.StudentDisplayByIdDto;
-import com.demo.MybatisApplication.dto.StudentsDisplayDto;
+import com.demo.MybatisApplication.dto.StudentAdditionDto;
+import com.demo.MybatisApplication.dto.StudentSubjectsDisplayDto;
+import com.demo.MybatisApplication.dto.StudentDisplayDto;
+import com.demo.MybatisApplication.exceptions.ResourceNotFoundException;
 import com.demo.MybatisApplication.model.StudentEntity;
 import com.demo.MybatisApplication.model.SubjectEntity;
 import com.demo.MybatisApplication.repository.StudentRepository;
+import com.demo.MybatisApplication.repository.SubjectRepository;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PathVariable;
 import com.demo.MybatisApplication.mapstruct.StudentMapper;
-import org.springframework.web.bind.annotation.RequestBody;
 
 import java.util.List;
 
 @Service
 public class StudentService {
     @Autowired
     private StudentRepository studentRepository;
-
     @Autowired
     private StudentMapper studentMapper;
+    @Autowired
+    private SubjectService subjectService;
 
-    public StudentDisplayByIdDto getStudentById(Long id){
+    public ResponseEntity<?> getStudentById(Long id){
         StudentEntity studentEntity = studentRepository.findStudentById(id);
-        return studentMapper.studentEntityToDisplayByIdDto(studentEntity);
+        if (studentEntity == null) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND)
+                    .body("Student with id " + id + " not found");
+        }
+
+        StudentDisplayDto studentDisplayDto = studentMapper.mapStudentEntityToStudentDisplayDto(studentEntity);
+        return ResponseEntity.ok().body(studentDisplayDto);
     }
 
-    public void assignSubjectsToStudent(Long studentId, List<SubjectEntity> subjects) {
-        studentRepository.updateSubjectsToStudent(studentId, subjects);
+    public ResponseEntity<String> assignSubjectsToStudent(Long studentId, List<Long> subjectIds) {
+
+        StudentEntity studentEntity = studentRepository.findStudentById(studentId);
+        if (studentEntity == null) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Student with id " + studentId + " not found");
+        }
+
+        if(subjectService.assignSubjectsToStudentHandling(subjectIds).equals("Done")){
```

Comment:
This if-else block is wrong. This method is being called 2 times one to evaluate if condition and 2nd time again in the else block.

This is how the flow could be- 
Call the getAllSubjectsByIds method.
Use the return subject entities to assign it to the student.
Use